kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/ResponsiveMongoClient.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.internal.db.mongo;
+
+import com.mongodb.client.MongoClient;
+import dev.responsive.kafka.internal.db.RemoteKVTable;
+import dev.responsive.kafka.internal.db.TableCache;
+import dev.responsive.kafka.internal.db.spec.BaseTableSpec;
+import java.util.concurrent.TimeoutException;
+
+public class ResponsiveMongoClient {
+
+  private final TableCache<RemoteKVTable> cache;
+  private final MongoClient client;
+
+  public ResponsiveMongoClient(final MongoClient client) {
+    this.client = client;
+    cache = new TableCache<>(spec -> new MongoKVTable(client, spec.tableName()));
+  }
+
+  public RemoteKVTable kvTable(final String name) throws InterruptedException, TimeoutException {
```

Comment:
It's not a big deal right now if things are working, but presumably we will eventually want the clients to implement a common interface and integrate into the new Spec/Operations structure, right?

Reply:
yup, that's definitely something I'd like to do